account_tax_expense_include/account_tax.py

```
@@ -30,3 +30,92 @@ class account_tax(orm.Model):
+                'taxes': []           # Taxes, see compute for the format
+            }
+        """
+
+        if not is_expense:
+            return super(account_tax, self).compute_all(
+                cr, uid, taxes, price_unit, quantity,
+                product=product, partner=partner, force_excluded=force_excluded
+            )
+
+        # By default, for each tax, tax amount will first be computed
+        # and rounded at the 'Account' decimal precision for each
+        # PO/SO/invoice line and then these rounded amounts will be
+        # summed, leading to the total amount for that tax. But, if the
+        # company has tax_calculation_rounding_method = round_globally,
+        # we still follow the same method, but we use a much larger
+        # precision when we round the tax amount for each line (we use
+        # the 'Account' decimal precision + 5), and that way it's like
+        # rounding after the sum of the tax amounts of each line
+        precision = self.pool.get('decimal.precision').precision_get(
+            cr, uid, 'Account')
+        tax_compute_precision = precision
+        if taxes and taxes[0].company_id.\
+                tax_calculation_rounding_method == 'round_globally':
+            tax_compute_precision += 5
+        totalin = totalex = round(price_unit * quantity, precision)
+        tin = []
+        tex = []
+        for tax in taxes:
+            if (tax.expense_include or tax.price_include)\
+                    and not force_excluded:
+                tin.append(tax)
+            else:
+                tex.append(tax)
+
+        tin = self.compute_inv(
+            cr, uid, tin, price_unit, quantity, product=product,
+            partner=partner, precision=tax_compute_precision)
+        for r in tin:
+            totalex -= r.get('amount', 0.0)
+        totlex_qty = 0.0
+        try:
+            totlex_qty = totalex / quantity
+        except:
+            pass
```

Please explain why you're swallowing all errors. It's better to specify the errors you're expecting to get.